modules/javafx.graphics/src/main/java/javafx/scene/layout/BackgroundImage.java

```diff
@@ -138,6 +150,68 @@ public BackgroundImage(
         hash = result;
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @throws NullPointerException {@inheritDoc}
+     * @since 24
+     */
+    @Override
+    public BackgroundImage interpolate(BackgroundImage endValue, double t) {
+        Objects.requireNonNull(endValue, "endValue cannot be null");
+
+        // We don't check equals(endValue) here to prevent unnecessary equality checks,
+        // and only check for equality with 'this' or 'endValue' after interpolation.
+        if (t <= 0) {
+            return this;
+        }
+
+        if (t >= 1) {
+            return endValue;
+        }
+
+        // BackgroundPosition and BackgroundSize are implemented such that interpolate() always returns
+        // the existing instance if the intermediate value is equal to the start value or the end value,
+        // which allows us to use an identity comparison in place of a value comparison to determine
+        // equality.
+        BackgroundPosition newPosition = this.position.interpolate(endValue.position, t);
+        BackgroundSize newSize = this.size.interpolate(endValue.size, t);
+        BackgroundRepeat newRepeatX, newRepeatY;
+        Image newImage;
+
+        if (t < 0.5) {
+            newRepeatX = this.repeatX;
+            newRepeatY = this.repeatY;
+            newImage = this.image;
+        } else {
+            newRepeatX = endValue.repeatX;
+            newRepeatY = endValue.repeatY;
+            newImage = endValue.image;
+        }
+
+        if (isSame(newImage, newRepeatX, newRepeatY, newPosition, newSize)) {
+            return this;
+        }
+
+        if (endValue.isSame(newImage, newRepeatX, newRepeatY, newPosition, newSize)) {
+            return endValue;
+        }
+
+        return new BackgroundImage(newImage, newRepeatX, newRepeatY, newPosition, newSize);
+    }
+
+    private boolean isSame(Image image,
+                           BackgroundRepeat repeatX,
+                           BackgroundRepeat repeatY,
+                           BackgroundPosition position,
+                           BackgroundSize size) {
+        return this.image == image
+            && this.repeatX == repeatX
+            && this.repeatY == repeatY
+            && this.position == position
+            && this.size == size;
```

Comment:
same comment on equals().

'size' comes from interpolate() on L178, so there is no guarantee as of its identity

Reply:
See my other comment. We _know_ that `BackgroundSize` returns existing instances if the intermediate value would be equal.